src/components/shared-settings/createSettings.tsx

```diff
@@ -17,7 +17,7 @@ function createInternalSettings<T extends browser.storage.StorageValue>(
         ready: boolean
         readonly readyPromise: Promise<void>
     }
-    let ready: () => void = () => {}
+    let ready: () => void = undefined!
```

Comment:
why is `undefined!`?

Reply:
By the execution order of Promise, there is no way to call the `() => {}` because it will be immediately replaced by the Promise resolved. Replace it is a no-op.

But if it really happened, throw an error `undefined is not a function` is better than eat the promise ready message sliently.